File: eshop_back/src/main/java/eshop/model/Personne.java
package eshop.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

@Entity
public class Personne {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;
	private String nom;
	private String prenom;
	
	public Personne() {}
	
	public Personne(String nom, String prenom) {
		this.nom = nom;
		this.prenom = prenom;
	}

	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public String getNom() {
		return nom;
	}

	public void setNom(String nom) {
		this.nom = nom;
	}

	public String getPrenom() {
		return prenom;
	}

	public void setPrenom(String prenom) {
		this.prenom = prenom;
	}

	@Override
	public String toString() {
		return "Personne [id=" + id + ", nom=" + nom + ", prenom=" + prenom + "]";
	}

	
	
	

}

File: ANGULAR/TYPESCRIPT/personne.ts

class Personne {

    constructor(private _nom: string, private _prenom?: string) { }

    public get nom(): string {
        return this._nom;
    }

    public set nom(value : string) {
        this._nom = value;
    }

}

let personne: Personne = new Personne("Le nom");

console.log(personne.nom);

personne.nom = "Nouveau nom";

console.log(personne.nom);
